util/types/compare.go

```diff
@@ -13,6 +13,9 @@
 
 package types
 
+// floatEpsilon is the acceptable error quantity when comparing two float numbers.
+const floatEpsilon float64 = 2.2204460492503131e-16
```

Comment:
`==` in go just compare memory contents of two float64, just like all other program languages. But I checked `Arg_comparator::set_compare_func` in `sql/item_cmpfunc.cc` in **MySQL**, that's much more complex for comparing two float64. I will check that later.